File: insertion/golang/insertionNomes.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"time"
)

func main() {
	input, err := os.Open("nomes 1.txt")
	if err != nil {
		panic(err)
	}
	defer input.Close()
	output, err := os.OpenFile("./insGoNomes.txt", os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
	if err != nil {
		panic(err)
	}
	defer input.Close()

	vetor := make([]string, 0)
	var comp, swap int

	scanner := bufio.NewScanner(input)
	num, err := strconv.Atoi(os.Args[1])
	for i := 0; i < num; i++ {
		var aux string
		scanner.Scan()
		aux = scanner.Text()
		vetor = append(vetor, aux)
	}

	// for _, v := range vetor {
	// 	fmt.Printf("%s, ", v)
	// }
	// println("\nsort\n")

	inicio := time.Now()
	for i := 0; i < len(vetor)-1; i++ {
		for j := i + 1; j > 0; j-- {
			comp++
			if vetor[j] < vetor[j-1] {
				var aux1 string
				aux1 = vetor[j]
				vetor[j] = vetor[j-1]
				vetor[j-1] = aux1
				swap++
			} else {
				break
			}
		}
	}
	fim := time.Now()

	// for _, v := range vetor {
	// 	fmt.Printf("%s, ", v)
	// }

	fmt.Fprintf(output, "%s,%v,%d,%d\n", os.Args[1], float64(fim.Sub(inicio).Seconds()), comp, swap)
}

File: insertion/cpp/insertionNomes.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <time.h>

int main(int argc, char *argv[2])
{
    std::ifstream input("nomes 1.txt");
    std::ofstream output("./insCppNomes.txt", std::ios_base::app);
    std::vector<std::string> vetor;
    std::string aux;
    unsigned long comp = 0, swap = 0;
    clock_t inicio, fim;

    for (int i = 0; i < atoi(argv[1]); i++)
    {
        std::getline(input, aux);
        vetor.push_back(aux);
    }

    inicio = clock();

    for (int i = 0; i < std::size(vetor) - 1; i++)
    {
        for (int j = i + 1; j > 0; j--)
        {
            comp++;
            if (vetor[j] < vetor[j - 1])
            {
                aux = vetor[j];
                vetor[j] = vetor[j - 1];
                vetor[j - 1] = aux;
                swap++;
            }
            else
                break;
        }
    }

    fim = clock();
    output << argv[1] << "," << float(((fim - inicio) + 0.0) / CLOCKS_PER_SEC) << "," << comp << "," << swap << "\n";

    return 0;
}